File: u1/Sample_Code/DemoTCPReceiver.java
import java.io.*;
import java.net.*;

public class DemoTCPReceiver{
	
	public static void main(String args[]){
		try{
			String _str; //declare variable for later use
			BufferedReader _reader;
			
			ServerSocket _server = new ServerSocket(12345); //prepare the ServerSocket which will create a Socket object when there is a new connection
			
			while(true){
				_str = ""; //reset variable whenever there is a new connection
				_reader = null;
				
				System.out.println("Waiting for connection in");
				Socket _socket = _server.accept(); //Waiting...
				
				System.out.printf("Connection with %s created%n", _socket.getInetAddress().getHostAddress()); //If this code run, that's mean there is a connection in
				
				_reader = new BufferedReader(new InputStreamReader(_socket.getInputStream()));//get the input stream from the socket and create a reader that read the stream data
				
				System.out.println("********** Message from sender **********");
				while((_str = _reader.readLine()) != null){//read the data from input stream
					System.out.println(_str);//process it
				}//repeat reading
				
				_socket.close();//or close the connection
				System.out.println("Connection closed");
			}//process next sender request
		}catch(IOException ioe){
			ioe.printStackTrace();
		}catch(Exception e){
			e.printStackTrace();
		}
	}
	
}

File: u1/Python/DemoTCPReceiver.py
import socket

def main():
	serversocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM) #create socket
	serversocket.bind(("", 12345)) #bind to port 12345
	serversocket.listen(1) #listen request from sender and wait
	print ('Waiting for connection in')
	
	while True:
		(sock, address) = serversocket.accept() #Receiver accept, create a socket
		print ('Connection with {addr} created'.format(addr=address[0]))
		
		print ('{a} Message from sender {a}'.format(a='*'*10))
		while True: #Repeating read
			line = ''
			line = sock.recv(1024).strip() #Receiver receive data from socket
			if line:
				print (line.decode('utf-8')) #process it
			else:
				break
		sock.close() # or close the connection
		print ('Connection closed')

if __name__ == "__main__":
	main()